### PR TITLE
gain fix

### PR DESCRIPTION
**Pull request by Andrey1994**
Signed-off-by: Andrey Parfenov <a1994ndrey@gmail.com>

### CODE REVIEW

File: OpenBCI_GUI/W_TimeSeries.pde [hunk 1]
```diff
@@ -273,6 +273,13 @@ class W_timeSeries extends Widget {
                     hsc.isVisible = false;
                     hardwareSettingsButton.setString("Hardware Settings");
                 } else{
+                    // if we change gains and other setting during the streaming there will be sync issue
+                    // force user to stop streaming first
+                    if (isRunning) {
+                        // todo better option to notify user? popup window doesnt work in processing....
+                        println("Need to pause streaming to configure the board.");
+                        return;
+                    }
```
Comment (daniellasry) [inline OpenBCI_GUI/W_TimeSeries.pde:282]: I'll make a quick popup message you can use

File: OpenBCI_GUI/Extras.pde [hunk 2]
```diff
@@ -308,26 +308,23 @@ class DataPacket_ADS1299 {
...
 class DataStatus {
     public boolean is_railed;
-    private int threshold_railed;
+    private double threshold_railed_uv;
     public boolean is_railed_warn;
-    private int threshold_railed_warn;
+    private double threshold_railed_warn_uv;
 
     DataStatus(int thresh_railed, int thresh_railed_warn) {
+        double eeg_scaler =  (4.5 / (pow (2, 23) - 1) / 24.0 * 1000000.);
```
[the remark this answers is not in the document]
Comment (Andrey1994) [reply OpenBCI_GUI/Extras.pde:316]: almost random, I copypasted this equation from brainflow's cyton code. Equation itself is fine but I am not sure that we need it here

**Comment by retiutut** [comment]
My primary comment here is that we should be very careful with this, as getting “un-railed” is already part of the learning curve for new users. We need this to be right, otherwise it could set back new users and internal testing. Thank you both for noticing that this needs to be addressed.😇

File: OpenBCI_GUI/BoardCyton.pde [hunk 3]
```diff
@@ -123,6 +123,19 @@ implements ImpedanceSettingsBoard, AccelerometerCapableBoard, AnalogCapableBoard
...
+    private Map<Integer, Double> scalers = new HashMap<Integer, Double>();
+    private Map<Character, Integer> gainCommandMap = new HashMap<Character, Integer>() {{
+        put('0', 1);
+        put('1', 2);
+        put('2', 4);
+        put('3', 6);
+        put('4', 8);
+        put('5', 12);
+        put('6', 24);
+    }};
```
[the remark this answers is not in the document]
Comment (Andrey1994) [reply OpenBCI_GUI/BoardCyton.pde:136]: like Char Map it looks better I think than smth like
```
char gain = smth_from_config_string;
gainArray[gain - '0']
```

File: OpenBCI_GUI/BoardCyton.pde [hunk 4]
```diff
@@ -148,10 +148,9 @@ implements ImpedanceSettingsBoard, AccelerometerCapableBoard, AnalogCapableBoard
...
     public BoardCyton() {
         super();
-        int[] exgChannels = getEXGChannels();
-        for (int i = 0; i < exgChannels.length; i++) {
-            // its not a bug, egChannels dont match channels in hardwaresettingscontroller
-            scalers.put(Integer.valueOf(i), Double.valueOf(1.0));
+        scalers = new double[getNumEXGChannels()];
+        for (int i = 0; i < scalers.length; i++) {
+            scalers[i] = 1.0;
         }
```
Comment (daniellasry) [inline OpenBCI_GUI/BoardCyton.pde]: You could use the helper Arrays.fill(scalers , 1.0);

**Pull request merged**